quadrat/sass/_header.scss

```diff
@@ -1,8 +1,39 @@
 .site-header {
-	.wp-block-group {
-		display: flex;
-		justify-content: space-between;
+	margin-bottom: var(--wp--custom--margin--vertical);
+
+	> .wp-block-group {
 		align-items: center;
+		display: flex;
+		flex-wrap: wrap;
+		justify-content: flex-start;
 		overflow: inherit;
+		padding: 60px 20px;
+
+		@include break-mobile() {
```

Comment:
I'm not sure about another way to handle the responsiveness besides media queries, but I've opened an issue here about spacing to discuss more generally: https://github.com/Automattic/themes/issues/3818